SabaCenter/Model/Repositories/LectureRepository.swift

```diff
@@ -0,0 +1,27 @@
+//
+//  LectureRepository.swift
+//  SabaCenter
+//
+//  Created by Mohsen Hosseini on 4/22/18.
+//  Copyright © 2018 Saba. All rights reserved.
+//
+
+import Foundation
+import RxSwift
+
+public class LectureRepository: ILectureRepository {
+    private let lectureService: ILectureService
+    private let lectureCache = [LectureItem]()
+
+    init(lectureService: LectureService) {
```

Comment:
this should be take the `ILectureService` interface as a parameter, not the concrete `LectureService` implementation